### PR TITLE
bda - getter function

### PR DESCRIPTION
**Pull request by zoometh**
first step to recover the ['bda' C14 database ](https://nakala.fr/10.34847/nkl.dde9fnm8)

added values/complete the files:

  - url_references.csv
  - variable_reference.csv
  - material_thesaurus.csv
  - country_thesaurus.csv

added 'bda' to get_c14data()

### CODE REVIEW

**Comment by dirkseidensticker** [comment]
@zoometh what a great starting point. I do not know how far you are to write the actual parser, but maybe you want to consider to have a look a the [parser for the euroevol database](https://github.com/ropensci/c14bazAAR/blob/master/R/get_euroevol.R). This database is also based on multiple individual files that need to be joined. 

If you need any assistance, I am there as well. Many thanks for your contribution already 👍

**Comment by zoometh** [comment]
@dirkseidensticker Yes I've started to write the `get_bda()` function inspired by the `get_euroevol()` function. I guess the `get_bda()` function will be ready within a week. If I have some issues, I'll let you know.

**Comment by zoometh** [comment]
@dirkseidensticker I've finally created the `get_bda` function. The joins between the 3 unzipped `.xlsx` files (Dates, Sites, Occupations) works but I was unable to check it with the c14bazAAR package since the `url_references.csv` file appears to be removed (temporally) from the package, maybe because of my previous commit.. This is the first time I used branches, and my skills with GitHub are still basics. Whatever, I can still help out.

**Comment by dirkseidensticker** [comment]
@zoometh we made some changes within #128 and the latest [v2 release](https://github.com/ropensci/c14bazAAR/tree/2.0.0). We removed the url table which one had to update on the remote. Could you pull the changes and add your url's to the new [db_info_table.csv](https://github.com/ropensci/c14bazAAR/blob/master/data-raw/db_info_table.csv) table? `get_db_url("[the database name]")` refers now to this table.

Removing the odd `url_references.csv` reference was something we wanted to do for some time, sorry that it intervened with your PR.

**Comment by zoometh** [comment]
I've just added a line in `db_info_table.csv` to register the `bda` database, [here](https://github.com/ropensci/c14bazAAR/pull/130/commits/8a452d783b813c35951fe9337d35a395f770cb74)

File: R/get_bda.R [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  td <- tempdir()
+
+  utils::download.file(
+    db_url,
+    temp,
+    mode="wb",
+    quiet = TRUE
+  )
+
+  unzip(temp, files="BDA-Table_Dates.xlsx", exdir=td, overwrite=TRUE)
```
Comment (dirkseidensticker) [inline R/get_bda.R:17]: add `utils::unzip(...`

File: R/get_bda.R [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  c14dates <- readxl::read_excel(paste0(td,"/BDA-Table_Dates.xlsx"))
+  unzip(temp, files="BDA-Table_Occupations.xlsx", exdir=td, overwrite=TRUE)
```
Comment (dirkseidensticker) [inline R/get_bda.R:19]: add `utils::unzip(...`

File: R/get_bda.R [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  c14occupa <- readxl::read_excel(paste0(td,"/BDA-Table_Occupations.xlsx"))
+  unzip(temp, files="BDA-Table_Sites.xlsx", exdir=td, overwrite=TRUE)
```
Comment (dirkseidensticker) [inline R/get_bda.R:21]: add `utils::unzip(...`

File: R/get_bda.R [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  c14sites <- readxl::read_excel(paste0(td,"/BDA-Table_Sites.xlsx"))
+
```
Comment (dirkseidensticker) [inline R/get_bda.R:23]: add 
```r
names(c14dates) <- gsub("\u00E9", "e", names(c14dates))
names(c14occupa) <- gsub("\u00E9", "e", names(c14occupa))
names(c14sites) <- gsub("\u00E9", "e", names(c14sites))
```
in order to replace non-ascii characters

File: R/get_bda.R [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+  bda <- c14dates %>%
+    dplyr::left_join(c14sites, by = c("id_site_associé" = "id_sites")) %>%
+    dplyr::left_join(c14occupa, by = c("id_occupation_liée" = "id_occupations_2019")) %>%
+    dplyr::transmute(
+      method = .data[["Méthode"]],
```
Comment (dirkseidensticker) [inline R/get_bda.R:28]: change `"Méthode"` into `"Methode"`

File: R/get_bda.R [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+      labnr = .data[["Code_labo"]],
+      c14age = .data[["Mesure"]],
+      c14std = .data[["Ecartype"]],
+      c13val = .data[["delta13"]],
+      site = .data[["Nom_site"]],
+      sitetype = .data[["Nature_site"]],
+      feature = .data[["num_couche"]],
+      period = .data[["période"]],
```
Comment (dirkseidensticker) [inline R/get_bda.R:36]: change `"période"` into `""periode"`

File: R/get_bda.R [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+      culture = .data[["culture"]],
+      material = .data[["Matériel_daté"]],
```
Comment (dirkseidensticker) [inline R/get_bda.R:38]: change `"Matériel_daté"` into `"Matèriel_date"

File: R/get_bda.R [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+      species = .data[["Matériel_daté_précision"]],
```
Comment (dirkseidensticker) [inline R/get_bda.R:39]: change `"Matériel_daté_précision"` into `"Materiel_date_precision"`

File: R/get_bda.R [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+      region = .data[["Région"]],
```
Comment (dirkseidensticker) [inline R/get_bda.R:40]: change `"Région"`into `"Region"`

File: R/get_bda.R [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+      country = .data[["Pays"]],
+      lat = .data[["Latitude"]],
+      lon = .data[["Longitude"]],
+      shortref = .data[["Ref_biblio"]],
+      comment = .data[["Fiabilité.x"]],
```
Comment (dirkseidensticker) [inline R/get_bda.R:45]: change `"Fiabilité.x"` into `"Fiabilite.x"`

File: R/get_bda.R [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+
+  unzip(temp, files="BDA-Table_Dates.xlsx", exdir=td, overwrite=TRUE)
+  c14dates <- readxl::read_excel(paste0(td,"/BDA-Table_Dates.xlsx"))
+  unzip(temp, files="BDA-Table_Occupations.xlsx", exdir=td, overwrite=TRUE)
+  c14occupa <- readxl::read_excel(paste0(td,"/BDA-Table_Occupations.xlsx"))
+  unzip(temp, files="BDA-Table_Sites.xlsx", exdir=td, overwrite=TRUE)
+  c14sites <- readxl::read_excel(paste0(td,"/BDA-Table_Sites.xlsx"))
+
+  bda <- c14dates %>%
+    dplyr::left_join(c14sites, by = c("id_site_associé" = "id_sites")) %>%
```
Comment (dirkseidensticker) [inline R/get_bda.R:25]: change `"id_site_associé"` into `"id_site_associe"`

File: R/get_bda.R [hunk 12]
```diff
@@ -0,0 +1,52 @@
...
+    dplyr::left_join(c14occupa, by = c("id_occupation_liée" = "id_occupations_2019")) %>%
```
Comment (dirkseidensticker) [inline R/get_bda.R:26]: change `"id_occupation_liée"` into `"id_occupation_liee"`

File: data-raw/url_reference.csv [hunk 13]
```diff
@@ -21,3 +21,4 @@ emedyd,2017,1,https://discovery.ucl.ac.uk/id/eprint/1570274/1/robertsetal17.zip
 katsianis,2020-08-20,1,https://rdr.ucl.ac.uk/ndownloader/files/23166314
 rapanui,2020-08-21,1,https://github.com/clipo/rapanui-radiocarbon/archive/master.zip
 mesorad,2020-09-01,1,https://github.com/eehh-stanford/price2020/raw/master/MesoRAD-v.1.1_FINAL_no_locations.xlsx
+bda,2021-03-23,1,https://api.nakala.fr/data/10.34847/nkl.dde9fnm8/189e04d917ffd68352a389006f357b58efda855e
```
Comment (dirkseidensticker) [inline data-raw/url_reference.csv:24]: line should be removed, as it is solved with #130

**Review by dirkseidensticker — CHANGES_REQUESTED** [review]
dear @zoometh , please excuse the long waiting time. I checked you parser, which looks (nearly) perfect! Congratulations and many thanks for your efforts from @nevrome and myself.

While running the checks, I noticed that you kept the french "é" in the column header. While this does not trigger an error, it flags a warning. Could you replace those characters?

Also, I noticed that you PR #130 was a bit redundant. Just keep in mind that when you add the url to the reference table, you would need to perform a "Clean and Rebuild" (in RStudio within the "More" menue within the "Build" tab. That triggers that the helper indices, like `db_info_table` are build again. Maybe we should add this to the README? @nevrome what do you think?

**Comment by zoometh** [comment]
Dear ***@***.****, ***@***.*** *I'm currently out of the office with
little access to the internet. Get back to work next Sunday (!). If that's
fine for you I will accomplish the required changes at this date. Let me
know,

**Comment by dirkseidensticker** [comment]
> ... Just keep in mind that when you add the url to the reference table, you would need to perform a "Clean and Rebuild" (in RStudio within the "More" menue within the "Build" tab. That triggers that the helper indices, like `db_info_table` are build again. Maybe we should add this to the README? @nevrome what do you think?

I missed this one yesterday: you need to run `data-raw/data_prep.R` to integrate the url added to the reference table (see no8 in [Adding database getter functions](https://github.com/ropensci/c14bazAAR#adding-database-getter-functions)).

**Comment by nevrome** [comment]
Yes - it's maybe a bit tricky that no5, 6 and 7 don't do anything without 8, but I didn't want to be redundant

**Review by zoometh — COMMENTED** [review]
I guess that's all fine. What will be the next step @dirkseidensticker ?

**Pull request closed without merging**